### PR TITLE
Confirm before sending to many recipients in to/cc

### LINKED ISSUE #4505: Confirm before sending to many recipients in `To` or `Cc`
### Feature Request

Sending mails to groups works great but users should be warned if they try to send an email with many recipients/groups or groups with many members in `To` or `Cc`. This should prevent leaking email addresses.

### PR DESCRIPTION
**Pull request by myrho**
Fixes #4505

I'm not sure about hard-coding the limit of to/cc recipients. Is there a better, configurable way?

### CODE REVIEW

File: lib/Exception/ManyRecipientsException.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * @copyright 2021 Christoph Wurst <christoph@winzerhof-wurst.at>
```
Comment (ChristophWurst) [inline lib/Exception/ManyRecipientsException.php:6]: it wasn't me

File: tests/Unit/Controller/AccountsControllerTest.php [hunk 2]
```diff
@@ -438,6 +439,26 @@ public function testSendingError() {
...
+	}
+
+	public function testSendingManyRecipientsCcError() {
+		$this->expectException(ManyRecipientsException::class);
+		$recipients = [];
+		for ($i = 0; $i < 11; $i++) {
+			$recipients[] = "$i@x.com";
+		}
+		$recipients = implode(',', $recipients);
+		$this->controller->send(13, 'sub', 'bod', '', $recipients, '');
```
Comment (ChristophWurst) [inline tests/Unit/Controller/AccountsControllerTest.php:459]: please add an empty line before this

most of our tests follow the AAA pattern: arrange, act, assert. The first few lines are where you arrange things, the final one is where you act.

File: src/components/Composer.vue [hunk 3]
```diff
@@ -226,6 +226,18 @@
 			{{ t('mail', 'Retry') }}
 		</button>
 	</div>
+	<div v-else-if="state === STATES.WARNING" class="emptycontent" role="alert">
+		<h2>{{ t('mail', 'Warning sending your message') }}</h2>
+		<p v-if="errorText">
+			{{ errorText }}
```
Comment (ChristophWurst) [inline src/components/Composer.vue:232]: this won't be translated. can we check on the error type and have a special case for ``ManyRecipientsError``?

File: src/components/Composer.vue [hunk 4]
```diff
@@ -278,6 +292,7 @@ const STATES = Object.seal({
 	SENDING: 2,
 	ERROR: 3,
 	FINISHED: 4,
+	WARNING: 5,
```
Comment (ChristophWurst) [inline src/components/Composer.vue:295]: I think that should be moved up and FINISHED should stay last

**Review by ChristophWurst — CHANGES_REQUESTED** [review]

File: src/components/Composer.vue (hunk 3 above)
Comment (myrho) [reply src/components/Composer.vue:232]: I did this by copy pasting and adapting L217-228. Matching for `ManyRecipientsError` happens in L740.

File: src/components/Composer.vue (hunk 4 above)
Comment (myrho) [reply src/components/Composer.vue:295]: So `WARNING: 4` and `FINISHED: 5`?

Comment (ChristophWurst) [reply src/components/Composer.vue:295]: yep :)

File: src/components/Composer.vue (hunk 3 above)
Comment (ChristophWurst) [reply src/components/Composer.vue:232]: my bad.

**Review by miaulalala — APPROVED** [review]
Looks good!

**Comment by myrho** [comment]
Ah sorry, I see. My bad!

**Pull request merged**